opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -2357,10 +2357,11 @@ namespace detail {
         }
 
         // compute PAV and PORV or every regions.
+        const V hydrocarbon = state.saturation[Oil].value() + state.saturation[Gas].value();
```

Comment:
If a phase is missing, this will crash, since value() will return a size zero array. Rather initialize to zero, then conditionally add each of the two saturations to it.